src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -84,16 +81,47 @@ public Task<FeatureDefinition> GetFeatureDefinitionAsync(string featureName)
                 throw new ArgumentException($"The value '{ConfigurationPath.KeyDelimiter}' is not allowed in the feature name.", nameof(featureName));
             }
 
-            EnsureInit();
-
             if (Interlocked.Exchange(ref _stale, 0) != 0)
             {
                 _definitions.Clear();
             }
 
-            //
-            // Query by feature name
-            FeatureDefinition definition = _definitions.GetOrAdd(featureName, (name) => ReadFeatureDefinition(name));
+            if (_definitions.TryGetValue(featureName, out FeatureDefinition definition))
+            {
+                return Task.FromResult(definition);
+            }
+
```

Comment:
I think to help readers follow what we have here it would be good to have something like

```
if (!TryGetMicrosoftSchemaFeatureDefinition(featureName, out definition))
{
    definition = GetDotnetSchemaFeatureDefinition(featureName);
}
```


Putting it all together
```
return Task.FromResult(
    _definitions.GetOrAdd(
        featureName,
        (name) =>
        {
            if (!TryGetMicrosoftSchemaFeatureDefinition(featureName, out definition))
            {
                definition = GetDotnetSchemaFeatureDefinition(featureName);
            }
        });
```

Reply:
How about

``` C#
return Task.FromResult(
    _definitions.GetOrAdd(
        featureName,
        (name) => GetMicrosoftSchemaFeatureDefinition(name) ?? GetDotnetSchemaFeatureDefinition(name)));
```

Both `GetMicrosoftSchemaFeatureDefinition` and `GetDotnetSchemaFeatureDefinition` method will return null if feature definition cannot be found.

Reply:
Sure